meta-lmp-base/recipes-containers/docker/files/0001-Fsync-layer-once-extracted-to-file-system.patch

```diff
@@ -0,0 +1,42 @@
+From 2df2205b02ebe4245bc056d5c2dca1bd2084c4b9 Mon Sep 17 00:00:00 2001
+From: Mike Sul <mike.sul@foundries.io>
+Date: Tue, 20 Dec 2022 11:45:13 +0100
+Subject: [PATCH] Fsync layer once extracted to file system
+
+- Sync an overall file system just after layer unpack while the control
+  flow is within the layer "change root".
+
+Signed-off-by: Mike <mike.sul@foundries.io>
+---
+ pkg/chrootarchive/diff_unix.go | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/pkg/chrootarchive/diff_unix.go b/pkg/chrootarchive/diff_unix.go
+index f0e9233ba3..01e60fcf22 100644
+--- a/src/import/pkg/chrootarchive/diff_unix.go
++++ b/src/import/pkg/chrootarchive/diff_unix.go
+@@ -8,9 +8,11 @@ import (
+ 	"encoding/json"
+ 	"flag"
+ 	"fmt"
++	"github.com/sirupsen/logrus"
+ 	"io"
+ 	"io/ioutil"
+ 	"os"
++	"os/exec"
+ 	"path/filepath"
+ 	"runtime"
+ 
+@@ -77,6 +79,9 @@ func applyLayer() {
+ 		fatal(err)
+ 	}
+ 
++	if syncErr := exec.Command("/bin/sync").Run(); syncErr != nil {
++		logrus.Errorf(">>> FIO: failed to sync the extracted layer directory")
++	}
```

Comment:
Isn't calling https://pkg.go.dev/syscall#Sync the same thing w/o the fork/exec and chance of an error?

Reply:
I think so, however `syscall.Sync()` doesn't return any error. I suppose it's not really important to catch the "sync" error since it's kind of best effort anyway.
I just wanted to make sure that the "sync" is actually executed without any error and check how it impact `docker pull` performance.

Reply:
Switched to `syscall.Sync()`, +- the same performance.

I think it's not important to catch the sync error, anyway the log output if printed in this spot is not visible because it's executed in "change root/root jail" env and its stderr is printed only if an overall extraction process exits with non-zero.

Reply:
this looks good - my point about catching the error was that there isn't one - the error fork/exec would produce would only be if /bin/sync didn't exist or something.